eth_account/account.py

```diff
@@ -229,6 +242,84 @@ def from_key(self, private_key):
         key = self._parsePrivateKey(private_key)
         return LocalAccount(key, self)
 
+    @combomethod
+    def from_mnemonic(self, mnemonic, passphrase="", account_path=ETHEREUM_DEFAULT_PATH):
+        """
+        :param str mnemonic: space-separated list of BIP39 mnemonic seed words
+        :param str passphrase: Optional passphrase used to encrypt the mnemonic
+        :param str account_path: Specify an alternate HD path for deriving the seed using
+            BIP32 HD wallet key derivation.
+        :return: object with methods for signing and encrypting
+        :rtype: LocalAccount
+
+        .. code-block:: python
+
+            >>> from eth_account import Account
+            >>> Account.enable_unaudited_hdwallet_features()
+            >>> acct = Account.from_mnemonic(
+              "coral allow abandon recipe top tray caught video climb similar prepare bracket "
+              "antenna rubber announce gauge volume hub hood burden skill immense add acid")
+            >>> acct.address
+            '0x9AdA5dAD14d925f4df1378409731a9B71Bc8569d'
+
+            # These methods are also available: sign_message(), sign_transaction(), encrypt()
+            # They correspond to the same-named methods in Account.*
+            # but without the private key argument
+        """
+        if not self._use_unaudited_hdwallet_features:
+            raise AttributeError(
+                "The use of the Mnemonic features of Account is disabled by default until "
+                "its API stabilizes. To use these features, please enable them by running "
+                "`Account.enable_unaudited_hdwallet_features()` and try again."
+            )
+        seed = seed_from_mnemonic(mnemonic, passphrase)
+        private_key = key_from_seed(seed, account_path)
+        key = self._parsePrivateKey(private_key)
+        return LocalAccount(key, self)
+
+    @combomethod
+    def create_with_mnemonic(self,
+                             passphrase="",
```

Comment:
These are all missing type hints.  Could you add them

@kclowes or @marcgarreau could one of you look into getting that enforced in CI (independent of this PR) so that we can ensure all new code is required to be typed.  Probably will require a little bit of work to finish off any parts of the codebase that aren't fully typed.

Reply:
I updated the type hints that were missing in what I did